src/main/java/Parser.java

```diff
@@ -0,0 +1,86 @@
+
+public class Parser {
+
+    private String command;
+    private String arguments;
+    public static final String SPACE_DELIMITER = " ";
+    public static final int START_INDEX = 0;
+
+    /**
+     * Parse input and separates the command from the arguments
+     *
+     * @param command the command line input to be parsed
+     */
+    public Parser(String command) {
+        command = command.strip();
+        if (command.isEmpty()) {
+            this.command = null;
+            arguments = null;
+        } else if (!command.contains(SPACE_DELIMITER)) {
+            this.command = command;
+            arguments = null;
+        } else {
+            int commandIndex = command.indexOf(SPACE_DELIMITER);
+            this.command = command.substring(START_INDEX, commandIndex).strip();
+            this.arguments = command.substring(commandIndex).strip();
+        }
+    }
+
+    /**
+     * Further splits the arguments based on a delimiter gives the first half the split string
+     *
+     * @param delimiter the delimiter to split the arguments by
+     * @return first half the string split delimiters
+     * @throws ArrayIndexOutOfBoundsException if the delimiter is not found in the string
+     */
+    public String getArgsBeforeString(String delimiter) throws ArrayIndexOutOfBoundsException {
+        int locationOfArgument = arguments.indexOf(delimiter);
+        if (locationOfArgument == -1) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return arguments.substring(0, locationOfArgument).trim();
```

Comment:
Perhaps you might want to add a final variable to store the value of 0 as START_OF_ARGUEMENT.